WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageListDialogHelper.kt

```diff
@@ -0,0 +1,89 @@
+package org.wordpress.android.viewmodel.pages
+
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.UNPUBLISHED_REVISION_DIALOG_LOAD_LOCAL_VERSION_CLICKED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.UNPUBLISHED_REVISION_DIALOG_LOAD_UNPUBLISHED_VERSION_CLICKED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.UNPUBLISHED_REVISION_DIALOG_SHOWN
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.RemoteId
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.ui.posts.PostUtils
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.helpers.DialogHolder
+
+private const val CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG = "CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG"
+private const val CONFIRM_DELETE_PAGE_DIALOG_TAG = "CONFIRM_DELETE_PAGE_DIALOG_TAG"
+
+class PageListDialogHelper(
+    private val showDialog: (DialogHolder) -> Unit,
+    private val analyticsTracker: AnalyticsTrackerWrapper
+) {
+    private var pageIdForAutosaveRevisionResolutionDialog: RemoteId? = null
+    private var pageIdForDeleteDialog: RemoteId? = null
+
+    fun showAutoSaveRevisionDialog(page: PostModel) {
+        analyticsTracker.track(UNPUBLISHED_REVISION_DIALOG_SHOWN)
+        val dialogHolder = DialogHolder(
+                tag = CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG,
+                title = UiStringRes(R.string.dialog_confirm_autosave_title),
+                message = PostUtils.getCustomStringForAutosaveRevisionDialog(page, true),
+                positiveButton = UiStringRes(R.string.dialog_confirm_autosave_restore_button),
+                negativeButton = UiStringRes(R.string.dialog_confirm_autosave_dont_restore_button)
+        )
+        pageIdForAutosaveRevisionResolutionDialog = RemoteId(page.remotePostId)
+        showDialog.invoke(dialogHolder)
+    }
+
+    fun showDeletePageConfirmationDialog(pageId: RemoteId, pageTitle: String) {
+        val dialogMessage = {
+            WordPress.getContext().getString(R.string.page_delete_dialog_message, pageTitle)
```

Comment:
> PageListDialogHelper is part of the viewmodel layer. I believe it shouldn't have access to the context. That's basically the main reason why we use UiString wrappers in this class. We might want to consider extending UIString so it supports parameters, wdyt?

Thanks for this! Modifying the `UiHelper` is a simple sensible approach. 

> I'm also not sure I follow the lambda approach. Why do we need to create a lambda just to invoke it once a few lines below?

Indeed, I was trying to avoid the long line of code within the `dialogHolder`. The solution above makes this cleaner. Thanks! 

Resolved in d5f27b7 8dd300b d276c95

Reply:
>Indeed, I was trying to avoid the long line of code within the dialogHolder.

I understand that part, but why we need the lambda. Wouldn't simply calling ```val dialogMessage = WordPress.getContext().getString(R.string.page_delete_dialog_message, pageTitle)``` without the parentheses be enough? If it was just a mistake, don't worry about it. I was just curious if it's a trick I'm not aware of.

>Resolved in d522cd2 d5f27b7 d276c95
In the above solution, I am passing the params as a List and then converting it to an array d5f27b7 when I realized it didn't behave as intended. Is this okay to do? I was going to use an array instead of a List but it's recommended that I implement equals & hashCode what do you think about this?

Using `Any` might bite us in the future. AFAIU  the params still need to be strings, so I'd try using `UIString`. Moreover, if someone passes UIString there, the current implementation will print something like `@ad87adf9`.
We might want to consider using [varargs](https://kotlinlang.org/docs/reference/functions.html#variable-number-of-arguments-varargs) instead of passing a list and converting it into a typed array. 

Wdyt?

Reply:
> If it was just a mistake, don't worry about it. I was just curious if it's a trick I'm not aware of.

Yes. It wasn't anything you weren't aware of 😄 

> Using Any might bite us in the future. AFAIU the params still need to be strings, so I'd try using UIString. Moreover, if someone passes UIString there, the current implementation will print something like @ad87adf9.

Thanks for expounding on this. I am going to utilize the `UiString` so that the params will always be strings. You know what I was thinking, I normally passed in different types so I was thinking about the support from that angle but now I am realizing those are still converted to strings with formatting. 

> We might want to consider using varargs instead of passing a list and converting it into a typed array.
Wdyt? 

Good point ! I considered doing this at first but what I realized was `varargs` [aren't supported in the constructor](https://stackoverflow.com/questions/31415093/constructor-call-in-kotlin-using-vararg). Let me know if there's another way to do this.

Also, I was going to pass it as an array but I end up using the `map` operator in the solution to transform each `UiString` to a string using the `UiHelper` method so that converts it back to a list anyway and it ends up being converted to a typed array so that the spread operator can be used. If you see a way this can be further simplified, let me know 😉 Resolved in 4b912f0

Reply:
> I considered doing this at first but what I realized was varargs aren't supported in the constructor. Let me know if there's another way to do this.

Ohh, I didn't know that. Thanks for clarification! I think using list is ok, especially since as you mentioned we use the `map`.

It all looks good now, I'm just curious. What is the `*` good for in `*uiString.params.map ...` ? I have never seen such usage. I know the usage when specifying generic type eg `<*>`, but I'm not sure what it does in front of a statement.

Reply:
TIL about the `*` - Destructuring operator. I understand the usage now. LGTM ;)!

Reply:
Awesome 😉